velox/exec/tests/SharedArbitratorTest.cpp

```diff
@@ -3628,131 +3865,68 @@ DEBUG_ONLY_TEST_F(SharedArbitrationTest, joinBuildSpillError) {
-
-  std::atomic<bool> stopped{false};
-
-  std::mutex mutex;
-  std::vector<std::shared_ptr<core::QueryCtx>> queries;
-  std::deque<std::shared_ptr<Task>> zombieTasks;
-
-  fakeOperatorFactory_->setAllocationCallback([&](Operator* op) {
-    if (folly::Random::oneIn(4)) {
-      auto task = op->testingOperatorCtx()->driverCtx()->task;
-      if (folly::Random::oneIn(3)) {
-        task->requestAbort();
-      } else {
-        task->requestYield();
-      }
+  for (const auto& capacities : std::vector<std::vector<uint64_t>>{
+           {16 * MB, 128 * MB},
+           {128 * MB, 16 * MB},
+           {128 * MB, 128 * MB},
+       }) {
+    auto totalCapacity = capacities[0];
+    auto queryCapacity = capacities[1];
+    setupMemory(totalCapacity);
+
+    FLAGS_velox_suppress_memory_capacity_exceeding_error_message = true;
+    const int numVectors = 8;
+    std::vector<RowVectorPtr> vectors;
+    fuzzerOpts_.vectorSize = 32;
+    fuzzerOpts_.stringVariableLength = false;
+    fuzzerOpts_.stringLength = 32;
+    vectors.reserve(numVectors);
+    for (int i = 0; i < numVectors; ++i) {
+      vectors.push_back(newVector());
     }
-    const size_t allocationSize = std::max(
-        kMemoryCapacity / 16, folly::Random::rand32() % kMemoryCapacity);
-    auto buffer = op->pool()->allocate(allocationSize);
-    return TestAllocation{op->pool(), buffer, allocationSize};
-  });
-  fakeOperatorFactory_->setMaxDrivers(numDrivers);
-  const std::string injectReclaimErrorMessage("Inject reclaim failure");
-  fakeOperatorFactory_->setReclaimCallback(
-      [&](MemoryPool* /*unused*/,
-          uint64_t /*unused*/,
-          MemoryReclaimer::Stats& /*unused*/) {
-        if (folly::Random::oneIn(10)) {
-          VELOX_FAIL(injectReclaimErrorMessage);
-        }
-        return false;
-      });
+    const int numDrivers = 4;
+    createDuckDbTable(vectors);
 
-  const int numThreads = 30;
-  const int maxNumZombieTasks = 128;
-  std::vector<std::thread> queryThreads;
-  for (int i = 0; i < numThreads; ++i) {
-    queryThreads.emplace_back([&, i]() {
-      DuckDbQueryRunner duckDbQueryRunner;
-      folly::Random::DefaultGenerator rng;
-      rng.seed(i);
-      while (!stopped) {
-        std::shared_ptr<core::QueryCtx> query;
-        {
-          std::lock_guard<std::mutex> l(mutex);
-          if (queries.empty()) {
-            queries.emplace_back(newQueryCtx());
-          }
-          const int index = folly::Random::rand32() % queries.size();
-          query = queries[index];
-        }
+    std::mutex mutex;
+    std::vector<std::shared_ptr<core::QueryCtx>> queries;
+    std::deque<std::shared_ptr<Task>> zombieTasks;
+
+    const int numThreads = 32;
+    const int maxNumZombieTasks = 8;
+    std::vector<std::thread> queryThreads;
+    queryThreads.reserve(numThreads);
+    for (int i = 0; i < numThreads; ++i) {
+      queryThreads.emplace_back([&, i]() {
         std::shared_ptr<Task> task;
```

Comment:
```
std::shared_ptr<Task> task;
try {
} catch {
}
```
Can you add to check at the end of each iterations that the arbitration has been triggered? Thanks!

Reply:
Got it, added `ASSERT_GT(arbitrator_->stats().numRequests, 0);` at the end of the testData loop as it does the `setupMemory` in each iteration.